dali/python/nvidia/dali/_multiproc/pool.py

```diff
@@ -559,6 +559,23 @@ def close(self):
             self.thread = None
 
 
+def create_shm_chunk_manager_for_group(group, shm_pool, keep_alive_queue_size,
+                                       min_initial_chunk_size, num_workers,
+                                       batch_size=None):
+    num_mini_batches = 1 if group.batch else num_workers
+    if group.bytes_per_sample_hint is None or batch_size is None:
+        initial_chunk_size = min_initial_chunk_size
+    else:
+        num_samples_per_mini_batch = (batch_size + num_mini_batches - 1) // num_mini_batches
+        initial_chunk_size = num_samples_per_mini_batch * group.bytes_per_sample_hint
```

Comment:
The user provides the hint per sample, but the statistics that we return are per minibatch. We also don't mention this fact in the documentation, I think I will probably be confusing and may result in confusion - I would reuse the statistics and got memory overallocation.

Reply:
I would suggest introducing `bytes_per_chunk_hint`, preferably hidden in a `parallel_options` disctionary, so we start grouping those specific options.